homeassistant/components/ted/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The TED integration."""
+from __future__ import annotations
+
+import logging
+
+import async_timeout
+import tedpy
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .const import COORDINATOR, DOMAIN, NAME, PLATFORMS
+from .coordinator import TIMEOUT, TedUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up TED from a config entry."""
+    config = entry.data
+    name = config[CONF_NAME]
+
+    async with async_timeout.timeout(TIMEOUT):
+        ted_reader = await tedpy.createTED(
+            config[CONF_HOST],
+            async_client=get_async_client(hass),
+        )
+
+    coordinator = TedUpdateCoordinator(hass, ted_reader=ted_reader, name=f"TED {name}")
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        COORDINATOR: coordinator,
+        NAME: name,
+    }
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    entry.async_on_unload(entry.add_update_listener(async_update_options))
```

Comment:
I couldn't see what this was for.

Reply:
It looks like `async_setup_platforms` was something used in entities when I first wrote this code but is not replaced with `async_forward_entry_setups(entry, PLATFORMS)`.

The `add_update_listener` I believe is to trigger a refresh of entities if the options change. I still see many integrations using this line.